### PR TITLE
Fix segfault when realizing to Dynamics after createFromStatesStorage()

### LINKED ISSUE #1455: Using a State from a StateTrajectory crashes on realizeDynamics call
Worked some with @aseth1 on this using MATLAB. If you get a State from a StateTrajectory and call `Model::realizeDynamics(state)`, a crash occurs with the top of the stack showing `SimTK::GeneralForceSubsystem::adoptForce`. If muscles set `appliesForce` to `false`, then no error occurs (and states can be looped to see the model move over multiple frames with the visualizer).

### PR DESCRIPTION
**Pull request by chrisdembia**
This PR fixes #1455.

This PR does the first of the two proposed fixes: requires the user to call `initSystem()` before invoking `createFromStatesStorage()`.

I had explored including the second change of invalidating the Instance-stage cache from the states in the trajectory. However, I ultimately decided against including this change. The reason is that I want users to be able to invoke `statesTraj.isConsistent()` immediately after `createFromStatesStorage()`, and `isConsistent()` requires the states to be realized to the Instance stage (for things like `getNQErr()`).

### CODE REVIEW

**Comment by chrisdembia** [comment]
@aseth1 @carmichaelong Could you review?

File: OpenSim/Simulation/StatesTrajectory.h [hunk 1]
```diff
@@ -321,6 +321,22 @@ class OSIMSIMULATION_API StatesTrajectory {
     };
 #endif
 
+    /** Thrown by createFromStatesStorage(). */
+    class ModelHasNoSystem : public Exception {
```
Comment (aseth1) [inline OpenSim/Simulation/StatesTrajectory.h]: This is a useful and more general Exception that could live with Model. Component has `ComponentHasNoSystem` but this is a good case for a Model specific one.

File: OpenSim/Simulation/StatesTrajectory.h [hunk 2]
```diff
@@ -414,6 +430,12 @@ class OSIMSIMULATION_API StatesTrajectory {
      * exactly reproduce results from the initial state trajectory; constraints
      * may not be satisfied, etc.
      *
+     * The states in the resulting trajectory will be realized to
+     * SimTK::Stage::Instance. You should not use the resulting trajectory with
+     * an instance of the model other than the you pass to this function; the
```
Comment (aseth1) [inline OpenSim/Simulation/StatesTrajectory.h]: "... other than the *one* you *passed* to this ..."

File: OpenSim/Simulation/StatesTrajectory.h [hunk 3]
```diff
@@ -414,6 +430,12 @@ class OSIMSIMULATION_API StatesTrajectory {
...
+     * state contains Instance-stage cache variables that point to force
```
Comment (aseth1) [inline OpenSim/Simulation/StatesTrajectory.h]: is it necessary to point to forces only? Any Component could have cache variables that point back into the Model.

Comment (aseth1) [inline OpenSim/Simulation/StatesTrajectory.h]: If it is just forces with raw pointers to `SimTK::Force`s we could disentangle so it works as expected. Perhaps post 4.0

Comment (chrisdembia) [inline OpenSim/Simulation/StatesTrajectory.h]: Ah...good question. I don't know...it seems that being able to cache pointers could have a performance benefit. What about PathPoints? Are they stored as cache variables?

I agree this is a larger issue to discuss, but as for this PR, do you want me to update the comment to be more general?

**Comment by chrisdembia** [comment]
Thank you @carmichaelong.

**Comment by chrisdembia** [comment]
I've addressed your comments. Now just waiting for the builds to pass.

**Comment by chrisdembia** [comment]
The builds have pass. Ready for review.

**Comment by chrisdembia** [comment]
Thank you!

**Pull request merged**